client/src/modules/libraries/index.jsx

```diff
@@ -28,18 +28,19 @@ class Libraries extends PureComponent {
     }
   }
 
-  getLibraries = () => {
+  getLibraries = async () => {
     const { fetchLibraries } = this.props;
 
-    this.setState({ pending: true });
-    this.pendingPromise = makeAbortablePromise(fetchLibraries());
-    this.pendingPromise.promise
-      .then(() => this.setState({ pending: false }))
-      .catch(error => {
-        if (!(error instanceof AbortPromiseException)) {
-          this.setState({ pending: false });
-        }
-      });
+    try {
+      this.setState({ pending: true });
+      this.pendingPromise = makeAbortablePromise(fetchLibraries());
+      await this.pendingPromise.promise;
+      this.setState({ pending: false });
```

Comment:
this should be set after the result from async action comes as resolved.

Reply:
The result variable isn't necessary. `this.setState({ pending: false });` will be invoke when `await this.pendingPromise.promise` finish its job (This is what `await` if for). If an error occurs catch will handle it and `this.setState({ pending: false });` won't be proceed. The result is needed when you expect that promise returns some value you want to use in nex steps